arbnode/transaction_streamer.go

```diff
@@ -262,50 +285,91 @@ func (s *TransactionStreamer) AddBroadcastMessages(feedMessages []*broadcaster.B
 	if len(feedMessages) == 0 {
 		return nil
 	}
-	startingSeqNum := feedMessages[0].SequenceNumber
+	broadcastStartPos := feedMessages[0].SequenceNumber
 	var messages []arbstate.MessageWithMetadata
-	endingSeqNum := startingSeqNum
+	broadcastAfterPos := broadcastStartPos
 	for _, feedMessage := range feedMessages {
-		if endingSeqNum != feedMessage.SequenceNumber {
-			return fmt.Errorf("invalid sequence number %v, expected %v", feedMessage.SequenceNumber, endingSeqNum)
+		if broadcastAfterPos != feedMessage.SequenceNumber {
+			return fmt.Errorf("invalid sequence number %v, expected %v", feedMessage.SequenceNumber, broadcastAfterPos)
 		}
 		if feedMessage.Message.Message == nil || feedMessage.Message.Message.Header == nil {
 			return fmt.Errorf("invalid feed message at sequence number %v", feedMessage.SequenceNumber)
 		}
 		messages = append(messages, feedMessage.Message)
-		endingSeqNum++
+		broadcastAfterPos++
 	}
 
 	s.insertionMutex.Lock()
 	defer s.insertionMutex.Unlock()
 
-	currentMessageCount, err := s.GetMessageCount()
+	var reorg bool
+	var err error
+	// Skip any messages already in the database
+	reorg, _, broadcastStartPos, messages, err = s.skipDuplicateMessages(0, broadcastStartPos, messages, true)
```

Comment:
I'd add a comment saying that we set the prevDelayedRead to 0 because it's only used to compute the output prevDelayedRead which we don't use

Reply:
:+1: